internal/support/include/beforefirstline.bi

```diff
@@ -0,0 +1,218 @@
+CONST _SECS_IN_MIN = 60, _MINS_IN_HOUR = 60, _HOURS_IN_DAY = 24
+CONST _DAYS_IN_WEEK = 7, _DAYS_IN_YEAR = 365
+'Derived time values
+CONST _SECS_IN_HOUR = _MINS_IN_HOUR * _SECS_IN_MIN
+CONST _SECS_IN_DAY = _HOURS_IN_DAY * _SECS_IN_HOUR
+CONST _SECS_IN_WEEK = _DAYS_IN_WEEK * _SECS_IN_DAY
+CONST _SECS_IN_YEAR = _DAYS_IN_YEAR * _SECS_IN_DAY
+CONST _SECS_IN_LEAPYEAR = _SECS_IN_YEAR + _SECS_IN_DAY
+CONST _MINS_IN_DAY = _HOURS_IN_DAY * _MINS_IN_HOUR
+CONST _MINS_IN_WEEK = _DAYS_IN_WEEK * _MINS_IN_DAY
+CONST _MINS_IN_YEAR = _DAYS_IN_YEAR * _MINS_IN_DAY
+CONST _MINS_IN_LEAPYEAR = _MINS_IN_YEAR + _MINS_IN_DAY
+CONST _HOURS_IN_WEEK = _DAYS_IN_WEEK * _HOURS_IN_DAY
+CONST _HOURS_IN_YEAR = _DAYS_IN_YEAR * _HOURS_IN_DAY
+CONST _HOURS_IN_LEAPYEAR = _HOURS_IN_YEAR + _HOURS_IN_DAY
+CONST _DAYS_IN_LEAPYEAR = _DAYS_IN_YEAR + 1
+CONST _WEEKS_IN_YEAR = _DAYS_IN_YEAR / _DAYS_IN_WEEK
+CONST _WEEKS_IN_LEAPYEAR = _DAYS_IN_LEAPYEAR / _DAYS_IN_WEEK
+
+'Binary size factors
+CONST _ONE_KB = 1024 '            ' 2^10
+CONST _ONE_MB = _ONE_KB * _ONE_KB ' 2^20
+CONST _ONE_GB = _ONE_MB * _ONE_KB ' 2^30
+CONST _ONE_TB = _ONE_GB * _ONE_KB ' 2^40
+CONST _ONE_PB = _ONE_TB * _ONE_KB ' 2^50
+CONST _ONE_EB = _ONE_PB * _ONE_KB ' 2^60
+
+'Sizes of elementary integer types (in bytes)
+CONST _SIZE_OF_BYTE = 1
+CONST _SIZE_OF_INTEGER = 2
+CONST _SIZE_OF_LONG = 4
+CONST _SIZE_OF_INTEGER64 = 8
+'Sizes of elementary floating point types (in bytes)
+CONST _SIZE_OF_SINGLE = 4
+CONST _SIZE_OF_DOUBLE = 8
+CONST _SIZE_OF_FLOAT = 32
+'Sizes of elementary pointer types (in bytes)
+$IF 32BIT THEN
+    CONST _SIZE_OF_OFFSET = 4
+$ELSE
+    CONST _SIZE_OF_OFFSET = 8
+$END IF
+
+'Limits of elementary integer types (min/max values)
+CONST _BIT_MIN` = -1, _BIT_MAX` = 0
+CONST _UBIT_MIN~` = 0, _UBIT_MAX~` = 1
+CONST _BYTE_MIN%% = -128, _BYTE_MAX%% = 127
+CONST _UBYTE_MIN~%% = 0, _UBYTE_MAX~%% = 255
+CONST _INTEGER_MIN% = -32768, _INTEGER_MAX% = 32767
+CONST _UINTEGER_MIN~% = 0, _UINTEGER_MAX~% = 65535
+CONST _LONG_MIN& = -2147483648, _LONG_MAX& = 2147483647
+CONST _ULONG_MIN~& = 0, _ULONG_MAX~& = 4294967295
+CONST _INTEGER64_MIN&& = -9223372036854775808, _INTEGER64_MAX&& = 9223372036854775807
+CONST _UINTEGER64_MIN~&& = 0, _UINTEGER64_MAX~&& = 18446744073709551615
+'Limits of elementary floating point types (min/max values)
+CONST _SINGLE_MIN! = -3.402823E+38, _SINGLE_MAX! = 3.402823E+38
+CONST _DOUBLE_MIN# = -1.797693134862315D+308, _DOUBLE_MAX# = 1.797693134862315D+308
+CONST _FLOAT_MIN## = -1.189731495357231765F+4932, _FLOAT_MAX## = 1.189731495357231765F+4932
+'Limits of elementary pointer types (min/max values)
+$IF 32BIT THEN
+    CONST _OFFSET_MIN& = -2147483648, _OFFSET_MAX& = 2147483647
+    CONST _UOFFSET_MIN~& = 0, _UOFFSET_MAX~& = 4294967295
+$ELSE
+    CONST _OFFSET_MIN&& = -9223372036854775808, _OFFSET_MAX&& = 9223372036854775807
+    CONST _UOFFSET_MIN~&& = 0, _UOFFSET_MAX~&& = 18446744073709551615
+$END IF
+
+'ASCII codes
+CONST _ASC_SPACE = 32
+CONST _ASC_EXCLAMATION = 33 '        !
```

Comment:
It's up to you but you could use `CHR$("!")` to define them rather than the number.